crates/wasmparser/src/validator/core.rs

```diff
@@ -513,6 +513,11 @@ impl Module {
                 }
                 Type::Func(t)
             }
+            crate::Type::Array(t) => {
+                // TODO check element type. Currently the check panics if the ref is not to a function type.
+                // self.check_revalue_type(t.element_type, features, types, offset)?;
```

Comment:
A reminder to handle this as it's probably one of the more important parts to handle before landing. (might be good to ensure there's `*.wast` tests for the various invalid cases too)

Reply:
Fixed.